### DISCUSSION TITLE
how does implicit resolution of nested node_modules folders work?

### DISCUSSION THREAD

**Discussion by debugwand** [webpack/webpack#18490] [Q&A]
I have found an issue where in certain conditions, webpack with sass-loader using the sass implementation works but sass-embedded does not, and initially raised the issue with example in sass-loader (https://github.com/webpack-contrib/sass-loader/issues/1203). Based on the below, I've some questions about how webpack resolution works for nested node_modules folders

Certain conditions are that you have a dependency in a nested `node_modules` folder of a dependency, and there is also a dependency called by more than one of the other dependencies (unrelated to the nested one) - the demo repo below has a basic mockup of this structure

As the findings in sass-loader where that what is returned from `prev` in sass-embedded were wrong, I put together a second example that used sass and sass-embedded without the use of loaders to help the owners of sass debug.

However, with this I found that neither sass nor sass-embedded are finding the nested `node_modules` folder without it being explicitly set in `includePaths` (which wasn't a big surprise). As the place in sass-loader where `prev` is showing up the wrong thing is in the `getWebpackImporter`, I think there may be something happening in webpack that is helping sass work but for whatever reason is not working for sass-embedded.

**What I am wondering:**
1) how/where does implicit resolution of nested `node_modules` folders work? (`node_modules` is mentioned in quite a lot of places in the codebase, there's nothing I spotted in the docs that goes into this level of internal detail)
2) might there be something happening that leads sass-embedded to think that the dependency that fails is called in via the other dependency rather than the main css file, but only when that other dependency is itself requiring a dependency that has already been required elsewhere? (the repo below sets up a mock of dependencies and will probably help to make sense of what this is)

Repos to demonstrate issues: 

Version with webpack, loaders and sass / sass-embedded, where sass works and sass-embedded does not for nested node_modules folders when there is a dependency called more than once elsewhere: https://github.com/debugwand/sass-embedded-bug

Version with no webpack or loaders, where neither sass nor sass-embedded works unless you list the nested node_modules folder explicitly in `includePaths` https://github.com/debugwand/sass-embedded-bug-noloaders

**Comment by alexander-akait** [comment]
> However, with this I found that neither sass nor sass-embedded are finding the nested node_modules folder without it being explicitly set in includePaths (which wasn't a big surprise). As the place in sass-loader where prev is showing up the wrong thing is in the getWebpackImporter, I think there may be something happening in webpack that is helping sass work but for whatever reason is not working for sass-embedded.

Because it is a bug on `sass-embedded` side, as you can see we don't have problems with JS/TS/etc projects

**Reply by alexander-akait** [reply]
Sorry, it is impossible to fix without dirty hacks

**Reply by alexander-akait** [reply]
Also you can try to setup alises for some borken resolutions, but if you code base is big, it will be diffucult
